signing/secp256k1/impl/src/test/java/tech/pegasys/signers/secp256k1/common/PasswordFileUtilTest.java

```diff
@@ -41,19 +40,17 @@ void canReadSpaceOnlyPasswordFromFile(@TempDir final Path tempDir) throws IOExce
     assertThat(PasswordFileUtil.readPasswordFromFile(passwordFile)).isEqualTo("   ");
   }
 
-  @Test
-  void emptyFileThrowsException(@TempDir final Path tempDir) throws IOException {
-    final Path passwordFile = Files.write(tempDir.resolve("password.txt"), new byte[0]);
-    assertThatExceptionOfType(SignerInitializationException.class)
-        .isThrownBy(() -> PasswordFileUtil.readPasswordFromFile(passwordFile))
-        .withMessage("Cannot read password from empty file: %s", passwordFile);
+  @ParameterizedTest
+  @ValueSource(strings = {"", "\n"})
+  void canReadPasswordFromEmptyStringFile(final String content, @TempDir final Path tempDir)
+      throws IOException {
+    final Path passwordFile = Files.write(tempDir.resolve("password.txt"), content.getBytes(UTF_8));
+    assertThat(PasswordFileUtil.readPasswordFromFile(passwordFile)).isEqualTo("");
   }
 
   @Test
-  void fileWithEolOnlyThrowsException(@TempDir final Path tempDir) throws IOException {
-    final Path passwordFile = Files.write(tempDir.resolve("password.txt"), "\n".getBytes(UTF_8));
-    assertThatExceptionOfType(SignerInitializationException.class)
-        .isThrownBy(() -> PasswordFileUtil.readPasswordFromFile(passwordFile))
-        .withMessage("Cannot read password from empty file: %s", passwordFile);
+  void canReadPasswordFromEmptyFile(@TempDir final Path tempDir) throws IOException {
+    final Path passwordFile = Files.write(tempDir.resolve("password.txt"), new byte[0]);
```

Comment:
Is empty string in the above test the same as this empty file case?

Reply:
Good point :)